akka-docs/src/test/java/jdocs/persistence/testkit/TestKitExamples.java

```diff
@@ -0,0 +1,180 @@
+          if (read.batch().nonEmpty()) {
+            ProcessingSuccess.getInstance();
+          } else {
+            return StorageFailure.create();
+          }
+        } else if (processingUnit instanceof WriteEvents) {
+          return ProcessingSuccess.getInstance();
+        } else if (processingUnit instanceof DeleteEvents) {
+          return ProcessingSuccess.getInstance();
+        } else if (processingUnit.equals(ReadSeqNum.getInstance())) {
+          return Reject.create();
+        }
+        // you can set your own exception
+        return StorageFailure.create(new RuntimeException("your exception"));
+      } else {
+        return ProcessingSuccess.getInstance();
+      }
+    }
+  }
+  // #set-event-storage-policy
+
+  // #set-snapshot-storage-policy
+  class SnapshotStoragePolicy implements ProcessingPolicy<SnapshotOperation> {
+
+    // you can use internal state, it doesn't need to be thread safe
+    int count = 1;
+
+    @Override
+    public ProcessingResult tryProcess(String persistenceId, SnapshotOperation processingUnit) {
+      // check the type of operation and react with success or with failure.
+      // if you return ProcessingSuccess the operation will be performed, otherwise not.
+      if (count < 10) {
+        count += 1;
+        if (processingUnit instanceof ReadSnapshot) {
+          ReadSnapshot read = (ReadSnapshot) processingUnit;
+          if (read.getSnapshot().isPresent()) {
+            ProcessingSuccess.getInstance();
+          } else {
+            return StorageFailure.create();
+          }
+        } else if (processingUnit instanceof WriteSnapshot) {
+          return ProcessingSuccess.getInstance();
+        } else if (processingUnit instanceof DeleteSnapshotsByCriteria) {
+          return ProcessingSuccess.getInstance();
+        } else if (processingUnit instanceof DeleteSnapshotByMeta) {
+          return ProcessingSuccess.getInstance();
+        }
+        // you can set your own exception
+        return StorageFailure.create(new RuntimeException("your exception"));
+      } else {
+        return ProcessingSuccess.getInstance();
+      }
+    }
+  }
+  // #set-snapshot-storage-policy
+
+}
+
+// #testkit-typed-usecase
+class SampleTest {
+
+  @ClassRule
+  public static final TestKitJunitResource testKit =
+      new TestKitJunitResource(
+          PersistenceTestKitPlugin.getInstance()
+              .config()
+              .withFallback(ConfigFactory.defaultApplication()));
+
+  PersistenceTestKit persistenceTestKit = PersistenceTestKit.create(testKit.system());
+
+  @Test
+  void test() {
+    ActorRef<Cmd> ref =
+        testKit.spawn(new YourPersistentBehavior(PersistenceId.ofUniqueId("some-id")));
+
+    Cmd cmd = new Cmd("data");
+    ref.tell(cmd);
+    Evt expectedEventPersisted = new Evt(cmd.data);
+
+    persistenceTestKit.expectNextPersisted("your-persistence-id", expectedEventPersisted);
+    persistenceTestKit.clearAll();
```

Comment:
The `pTK.clearAll()` only runs when the test completes successfully. This can cause a cascading failure of tests. Instead, we should suggest using `pTK.clearAll()` in a `@Before` method.

Reply:
good catch!